pkg/apis/flowcontrol/types.go

```diff
@@ -0,0 +1,298 @@
+	// Rules can either apply to API resources (such as "pods" or "secrets") or non-resource URL paths (such as "/api"),  but not both.
+	// +optional
+	NonResourceURLs []string
+}
+
+// FlowSchemaStatus represents the current state of a flow-schema.
+type FlowSchemaStatus struct {
+	// Current state of flow-schema.
+	Conditions []FlowSchemaStatusCondition
+}
+
+// FlowSchemaStatusCondition describes conditions for a flow-schema.
+type FlowSchemaStatusCondition struct {
+	// Type is the type of the condition.
+	Type FlowSchemaStatusConditionType
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
+	Status ConditionStatus
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time
+	// Unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string
+	// Human-readable message indicating details about last transition.
+	Message string
+}
+
+// FlowSchemaStatusConditionType is a valid value for FlowSchemaStatusCondition.Type
+type FlowSchemaStatusConditionType string
+
+// FlowSchemaStatusConditionStatus is a valid value for FlowSchemaStatusCondition.Status
+type FlowSchemaStatusConditionStatus string
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RequestPriority represents the configuration of a priority level.
+type RequestPriority struct {
+	metav1.TypeMeta
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta
+	// Specification of the desired behavior of a request-priority.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status
+	// +optional
+	Spec RequestPrioritySpec
+	// Current status of a request-priority.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status
+	// +optional
+	Status RequestPriorityStatus
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RequestPriorityList is a list of RequestPriority objects.
+type RequestPriorityList struct {
+	metav1.TypeMeta
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta
+	// Items is a list of request-priorities.
+	Items []RequestPriority
+}
+
+type RequestPrioritySpec struct {
+	// Exempted defines whether the priority level is exempted or not.  There should be at most one exempt priority level.
+	// Being exempt means that requests of that priority are not subject to concurrency limits (and thus are never queued)
+	// and do not detract from the concurrency available for non-exempt requests. In a more sophisticated system, the
+	// exempt priority level would be the highest priority level. It's defaulted to false and can only be true in the
```

Comment:
if this is the case, why is it a bool?

Reply:
i removed the doc and keep the field a pointer. the exemption should always be false except for some system presets, so making this a pointer make trim the yaml content a bit and make it looks a bit simpler.

Reply:
Huh?  Removing the pointer makes it simpler.  There are only two possibilities here, and the zero value for `bool` is the right default.

Reply:
now it's a boolean instead of a pointer.